packages/cursorless-vscode-core/StatusBarItem.ts

```diff
@@ -0,0 +1,35 @@
+import * as vscode from "vscode";
+
+const DEFAULT_TEXT = "$(cursorless-icon) Cursorless";
+
+export class StatusBarItem {
+  private statusBarItem?: vscode.StatusBarItem;
+
+  private constructor() {
+    // empty
+  }
+
+  static create(showQuickPickCommandId: string) {
+    const statusBarItem = new StatusBarItem();
+    statusBarItem.init(showQuickPickCommandId);
+    return statusBarItem;
+  }
+
+  private init(showQuickPickCommandId: string) {
```

Comment:
i've started using this pattern for things that have a required `init` step that should be run after construction: make the constructor private and then expose a static `create` function that constructs object and calls init

Reply:
Total agree!